sdk/core/abort-controller/README.md

```diff
@@ -1,61 +1,81 @@
+with the `AbortController` built into modern browsers and the `AbortSignal` used by `fetch`.
+These classes are provided for use with APIs in the Azure SDK for JavaScript that accept an `AbortSignalLike`
+parameter to cancel an operation.
 
 ## Getting started
 
 ### Installation
- - Installing this library
+
+- Installing this library
+
 ```
 npm install @azure/abort-controller
 ```
 
 ## Key Concepts
-This library contains the `Aborter` class which is used by the Azure SDK for JavaScript to abort pending
-work based on timeouts or other signales.
-`Aborter` is compatible with the `AbortSignal` used by the `fetch` API built into modern browsers.
+
+Use the `AbortController` to create an `AbortSignal` which can then be passed to Azure SDK operations to cancel
+pending work. The `AbortSignal` can be accessed via the `signal` property on an instantiated `AbortController`.
+An `AbortSignal` can also be returned directly from a static method, e.g. `AbortController.timeout(100)`.
+that is cancelled after 100 milliseconds.
+
+Calling `abort()` on the instantiated `AbortController` updates the associated `AbortSignal`.
+Any subsequent calls to `abort()` on the same controller will have no effect.
+
+The `AbortSignal.none` static property returns an `AbortSignal` that can not be aborted.
+Provde this to an Azure SDK method that requires an `abortSignal` if you don't wish to add custom abort logic.
+
+`AbortSignals` can also be linked so that when `abort()` is called on parent signal,
+it also fires on all child signals.
+This relationship is one-way, meaning that a parent signal can affect a child signal, but not the other way around.
+To create a parent-child relationship, pass in any parent signals to the `AbortController` constructor.
 
 ## Examples
 
 ### Example 1 - basic usage
 
 ```js
-import { Aborter } from "@azure/abort-controller";
+import { AbortController } from "@azure/abort-controller";
 
-const signal = new Aborter();
-doAsyncWork({ abortSignal: signal });
+const controller = new AbortController();
+doAsyncWork({ abortSignal: controller.signal });
 
 // at some point later
-signal.abort();
+controller.abort();
 ```
 
 ### Example 2 - Aborting with timeout
 
 ```js
-import { Aborter } from "@azure/abort-controller";
+import { AbortController } from "@azure/abort-controller";
 
-const signal = Aborter.timeout(1000);
+const signal = AbortController.timeout(1000);
 doAsyncWork({ abortSignal: signal });
 ```
 
 ### Example 3 - Aborting sub-tasks
 
 ```js
-import { Aborter } from "@azure/abort-controller";
+import { AbortController } from "@azure/abort-controller";
 
-const allTasksSignal = new Aborter();
+const allTasksController = new AbortController();
 
-const subTask1 = allTasksSignal.withValue("key1", "value1");
-const subtask2 = allTasksSignal.withValue("key2", "value2");
+const subTask1 = new AbortController(allTasksController.signal);
+const subtask2 = new AbortController(allTasksController.signal);
```

Comment:
How can we combine a parent signal with timeout?

Reply:
I'll add an example.

Reply:
Actually, I thought you meant how to support the case where the parent signal had a timeout, but are you looking for how to call AbortController.timeout() with parent signals as well?

I can add parentSignals as the last parameter to the `AbortController` static methods. Otherwise, if you wanted to use `AbortController.timeout()` for a subtask, as well as a parent signal, you'd need to do something like this:

```typescript
async function doStuff(parentSignal?: AbortSignalLike) {
  const timeout = AbortController.timeout(100);
  
  const subTaskController = new AbortController(parentSignal, timeout.signal);

}
```

/cc @bterlson

Reply:
> are you looking for how to call AbortController.timeout() with parent signals as well?

Yes. Looking for a case when timeout is on the child